packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -203,54 +202,36 @@ export function warnAboutNodeManifestMappingProblems({
   }
 }
 
-/**
- * Retrieves the content digest of a page-data.json file for use in creating node manifest files.
- */
-export async function getPageDataDigestForPagePath(
-  pagePath?: string,
-  directory?: string
-): Promise<string | null> {
-  if (
-    // if no page was created for the node we're creating a manifest for, there wont be a page path.
-    !pagePath ||
-    // we only add page data digests to node manifests in production because page-data.json may not exist in development.
-    (process.env.NODE_ENV !== `production` && process.env.NODE_ENV !== `test`)
-  ) {
-    return null
-  }
-
-  try {
-    const publicDirectory = path.join(
-      directory || store.getState().program.directory,
-      `public`
-    )
-    const pageData = await readPageData(publicDirectory, pagePath)
-
-    const pageDataDigest = createContentDigest(pageData)
-
-    return pageDataDigest
-  } catch (e) {
-    reporter.warn(
-      `No page-data.json found for ${pagePath} while processing node manifests.`
-    )
-
-    return null
-  }
-}
-
+let warnOnceAboutNodeManifestMappingProblems
```

Comment:
Instead of writing your own function you should reuse `reportOnce`: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/report-once.ts

I'd also only stick with the `verbose` mode and not add any additional env vars. You can check for verbose mode with

```js
process.env.gatsby_log_level === `verbose`
```

Reply:
Oh awesome, I didn't know that util existed! Though, I'm not sure if `reportOnce` makes sense here because we are passing an object not a string here. I'm going to try to rework the way we're doing logging here so it's going to change from what it is now. I'll run it by you when I'm done with it here!